### PR TITLE
Refactor(eos_designs): Wildcard dict to list for connected_endpoints

### PR DESCRIPTION
**Pull request by ghost**
## Data model key

`<connected_endpoints_keys.key>`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1: Claus
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Tony
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/doc/custom-structured-configuration-v4.0.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ All relevant `structured_config` sections will be merged.
 
 ```yaml
 < connected_endpoints_keys.key >:
-  < endpoint_1 >:
+  - endpoint: < endpoint_1 >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/custom-structured-configuration-v4.0.md]: please make this `name`

File: ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2 [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {# Leaf connected_endpoint interfaces #}
 ethernet_interfaces:
 {% for endpoint_key in connected_endpoints | arista.avd.natural_sort %}
-{%     for connected_endpoint in connected_endpoints[endpoint_key] | arista.avd.natural_sort %}
-{%         for adapter in connected_endpoints[endpoint_key][connected_endpoint].adapters %}
+{%     for connected_endpoint in connected_endpoints[endpoint_key] | arista.avd.convert_dicts('endpoint') | arista.avd.natural_sort %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2]: These templates are consuming data from the `logic.j2`, so you should only implement the conversion in logic. You don't need to change the output of logic.j2: `connected_endpoints`.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2 (hunk 2 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2]: Fixed

File: ansible_collections/arista/avd/roles/eos_designs/doc/connected-endpoints-v4.0.md [hunk 3]
```diff
@@ -395,7 +395,7 @@ Active/Active multihoming connections:
 
 ```yaml
 servers:
-  server01:
+  - endpoint: server01
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/connected-endpoints-v4.0.md]:
```suggestion
  - name: server01
```
same elsewhere

File: ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/logic.j2 [hunk 4]
```diff
@@ -1,10 +1,9 @@
 {% for connected_endpoints_key in connected_endpoints_keys %}
-{%     set switch_connected_endpoints = hostvars[inventory_hostname][connected_endpoints_key] | arista.avd.default({}) %}
+{%     set switch_connected_endpoints = hostvars[inventory_hostname][connected_endpoints_key] | arista.avd.convert_dicts('name') | arista.avd.default({}) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/logic.j2]:
```suggestion
{%     set switch_connected_endpoints = hostvars[inventory_hostname][connected_endpoints_key] | arista.avd.convert_dicts('name') | arista.avd.default([]) %}
```

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**